From 3df5f142d2a68ae137e68c19a36f59e5b58c8385 Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Thu, 14 Aug 2014 14:42:04 -0400
Subject: [ethernet] e1000e: Fix not generating an error on invalid load parameter

Message-id: <1408027348-9610-11-git-send-email-jogreene@redhat.com>
Patchwork-id: 87707
O-Subject: [RHEL7.1 PATCH 10/34] e1000e: Fix not generating an error on invalid load parameter
Bugzilla: 1091119
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Valid values for InterruptThrottleRate are 10-100000, or one of
0, 1, 3, 4.  '2' is not valid.  This is a legacy from the branching
from the e1000 driver code that e1000e was based from.

Prior to this patch, if the e1000e driver  was loaded with a forced
invalid InterruptThrottleRate of '2', then no throttle rate would be
set and no error message generated.

Now, a message will be generated that an invalid value was used and the
value for InterruptThrottleRate will be set to the default value.

Signed-off-by: Dave Ertman <davidx.m.ertman@intel.com>
Tested-by: Jeff Pieper <jeffrey.e.pieper@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit 5bb731760810b30d67096cbdded96addba4f1292)

BZ: 1091119
Testing: Acceptable, see BZ
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7834849
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/e1000e/param.c b/drivers/net/ethernet/intel/e1000e/param.c
index 01797b7..d0ac0f3 100644
--- a/drivers/net/ethernet/intel/e1000e/param.c
+++ b/drivers/net/ethernet/intel/e1000e/param.c
@@ -374,6 +374,12 @@ void e1000e_check_options(struct e1000_adapter *adapter)
      "%s set to dynamic mode\n", opt.name);
    adapter->itr = 20000;
    break;
+  case 2:
+   dev_info(&adapter->pdev->dev,
+     "%s Invalid mode - setting default\n",
+     opt.name);
+   adapter->itr_setting = opt.def;
+   /* fall-through */
   case 3:
    dev_info(&adapter->pdev->dev,
      "%s set to dynamic conservative mode\n",
-- 
1.7.1